libs/mock-block-dock/src/datastore/use-mock-datastore.ts

```diff
@@ -220,13 +225,47 @@ export const useMockDatastore = (
                   }
                 : undefined;
 
-            const updatedEntity: Entity = {
+            /**
+             * @todo - this is 'safe' because `getEntityRevisionFromSubgraph` returns the object from within the
+             *   `Subgraph`. If it did something like spreading and creating a new object then this wouldn't mutate
+             *   the vertex in the original graph. Are we happy with this, or should we do the messy code to refetch
+             *   it here again so we can be sure we're updating the object in the subgraph.
+             */
+            const currentTime = new Date().toISOString();
+
+            // Update the old entity revision's end times
+            const updatedOldRevisionEndBound = {
+              kind: "exclusive",
+              limit: currentTime,
+            } as const;
+
+            currentEntity.metadata.temporalVersioning.decisionTime.end =
+              updatedOldRevisionEndBound;
+            currentEntity.metadata.temporalVersioning.transactionTime.end =
+              updatedOldRevisionEndBound;
```

Comment:
We shouldn't do this because it is mutating an object in React state, which should be immutable (it might lead to weird bugs if other state updates happen around the same time)

I would do this at the top
```ts
const newSubgraph = JSON.parse(JSON.stringify(currentGraph));
```
then mutate `newSubgraph` as needed. the current assignment of `newSubgraph` below can be removed. Where we just return `currentGraph` unmodified that can stay the same, as it will skip a re-render.

The other option is to introduce a library like [immer](https://immerjs.github.io/immer/) but it seems like overkill for this one change.

Reply:
Ah bugger. I'll do as you suggest for now but it's really not fab to have to serialize and deserialize an entire datastore every time there's an update 🙃.

So, turns out we were already doing it.. I took a stab at fixing it without going overboard on unergonomic code
[Stop mutating react state](https://github.com/blockprotocol/blockprotocol/pull/931/commits/afd5379769edf8ef9c896a0be2a1dd0720507c5e)